src/TelemetryChannels/ServerTelemetryChannel/Shared/SamplingTelemetryProcessor.cs

```diff
@@ -139,72 +157,145 @@ public string IncludedTypes
         private ITelemetryProcessor Next { get; set; }
 
         /// <summary>
+        /// Initializes this processor using the correct telemetry pipeline configuration.
+        /// </summary>
+        /// <param name="configuration"></param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            this.metricManager = (configuration == null)
+                                        ? new MetricManager()
+                                        : new MetricManager(new TelemetryClient(configuration));
+        }
+
+        /// <summary>
         /// Process a collected telemetry item.
         /// </summary>
         /// <param name="item">A collected Telemetry item.</param>
         public void Process(ITelemetry item)
         {
-            if (this.SamplingPercentage < 100.0 - 1.0E-12)
+            double samplingPercentage = this.SamplingPercentage;
+
+            // If sampling rate is 100% we log the sapling rate as a metric and do nothing else:
+            if (samplingPercentage >= 100.0 - 1.0E-12)
             {
-                var samplingSupportingTelemetry = item as ISupportSampling;
+                
+                TrackSamplingRate(100.0);
+                this.Next.Process(item);
+                return;
```

Comment:
nitpick: They all call this.Next.Process(item) followed by return, can this be shortened up with else statements?